### ISSUE TITLE
Heartbeat continues after fail open lock is released

### ISSUE THREAD

**Issue by Jacob-Lynch** [tristanls/dynamodb-lock-client#1]
Hey, I appreciate you creating this library!  It helped me quickly get started with dynamodb locks.

Anyway, there is a bug that occasionally occurs where the following steps happen in this order:

- A heartbeat refreshLock function call happens and a dynamodb put call is made to refresh the lock.
- While waiting for the response from dynamodb a lock.release() function call is made.
- clearTimeout is called on the setTimeout's return value for the last heartbeat.

At this point the lock is released and the heartbeat is stopped.  But then...

- The dynamodb put callback fires and calls setTimeout to refresh the lock again in the future, causing the heartbeat to start back up.

Now the heartbeat continues indefinitely and keeps the process alive.  I run this inside of an AWS lambda and it keeps the lambda alive the full 300 seconds.

I tried a simple fix of adding:

`self._released = true;`

to Lock.prototype.release and then inside the refreshLock function right before calling setTimeout:

```
if(!self._released) {
  self._heartbeatTimeout = setTimeout(refreshLock, self._heartbeatPeriodMs);
}
```

This has prevented the heartbeat from turning back on after the lock is released.

Thanks again!

**Comment by tristanls** [comment]
Thank you very much for you trying this out and reporting the problem! Your problem description was straightforward to follow.

I implemented your solution and published [0.2.2](https://github.com/tristanls/dynamodb-lock-client/releases/tag/v0.2.2).

Thank you again!

**Comment by Jacob-Lynch** [comment]
Hey, thank you very much for the quick fix and release!
